home/home_3.1/unittest.swift

```diff
@@ -0,0 +1,65 @@
+import XCTest
+
+func calculateValue(x: Double, y: Double, z: Double) -> Double {
+    let product = x * y * z
+
+    if product != 0 {
+        return 1 / product
+    } else {
+        let sum = x + y + z
+
+        if sum != 0 {
+            return 1 / sum
+        } else {
+            return x + (y + 1) * (z - 1)
+        }
+    }
+}
+
+class CalculateValueTests: XCTestCase {
+
+    func testNonZeroProduct() {
+        let x = 2.0
+        let y = 3.0
+        let z = 4.0
+        let expectedResult = 1 / (x * y * z)
+        let result = calculateValue(x: x, y: y, z: z)
+        XCTAssertEqual(result, expectedResult, accuracy: 0.00001)
+    }
+
+    func testZeroProductNonZeroSum() {
+        let x = 2.0
+        let y = -1.0
+        let z = -2.0
+        let expectedResult = 1 / (x + y + z)
+        let result = calculateValue(x: x, y: y, z: z)
+        XCTAssertEqual(result, expectedResult, accuracy: 0.00001)
```

Comment:
you should cleanup this code duplication related to this hardcoded magic number. Just introduce a member with a good name like doubleAccuracy or doubleComparisonAccuracy.
In TDD cleanup stage means that you should review and clean all the code for both tests and the prod code - quality of test code has the same importance as the quality of prod code